plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetNotFoundException.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.google.sheets;
+
+import io.trino.spi.connector.NotFoundException;
+
+import static java.lang.String.format;
+
+public class SheetNotFoundException
+        extends NotFoundException
+{
+    public SheetNotFoundException(String sheetExpression)
+    {
+        super(format("Sheet '%s' not found", sheetExpression));
```

Comment:
This exception only gets thrown if a google sheet is removed or modified in a way that causes failure during processing of a query. 

Otherwise, this exception will be thrown when attempting to originally read the sheet range:
`TrinoException(SHEETS_UNKNOWN_TABLE_ERROR, "Sheet expression not found for table " + tableName)`

That is difficult to integration test with an actual sheet, but I can at least exercise this code path with a unit test.

Reply:
No need to add unit tests in my opinion.